crypto/pkcs11key/pkcs11key.go

```diff
@@ -128,71 +208,78 @@ func New(module, slot, pin, privLabel string) (ps *PKCS11Key, err error) {
 		}
 	}
 	if !gotModulus || !gotExponent {
-		ps.Destroy()
-		return
+		return noKey, errors.New("public key missing either modulus or exponent")
 	}
-	ps.publicKey = rsa.PublicKey{
+	return rsa.PublicKey{
 		N: n,
 		E: e,
-	}
-
-	return
+	}, nil
 }
 
-// Destroy tears down a PKCS11Key.
-//
-// This method must be called before the PKCS11Key is GC'ed, in order
-// to ensure that the PKCS#11 module itself is properly finalized and
-// destroyed.
-//
-// The idiomatic way to do this (assuming no need for a long-lived
-// signer) is as follows:
-//
-//   ps, err := NewPKCS11Signer(...)
-//   if err != nil { ... }
-//   defer ps.Destroy()
+
+// Destroy tears down a PKCS11Key by closing the session. It should be
+// called before the key gets GC'ed, to avoid leaving dangling sessions.
+// NOTE: We do not want to call module.Logout here. module.Logout applies
```

Comment:
Maybe put this not in the doc comments, but inside the function for implementers coming later. It's a little confusing.